### ISSUE TITLE
Bluesky can't detect a post's language reliably enough

### ISSUE THREAD

**Issue by ssg** [bluesky-social/social-app#7260]
### Steps to Reproduce

1. Draft a new post while non-English is selected: type this poem by William Wordsworth for example, but it works with many other texts. "never did sun more beautifully steep, in its first splendour rock or valley or hill. ne'er saw i, never felt a calm so deep. the river glideth in its own sweet will. dear god! many houses seem asleep and all that mighty heart is lying still!"
3. No notification about "Are you writing this in English?"

### What platform(s) does this occur on?

iOS, Web (Desktop)

### What version of the app are you using?

1.96.2

### Additional Information

I understand that this is because "lande", the package used for language detection, is quite unreliable. But it's obviously insufficient for Bluesky. This is significant because picking the wrong language while posting means loss of most of the reach along with loss of followers (due to the noise created for people who don't speak the language). Maybe the whole multilingual UX needs to be worked on, I'm not sure.

**Comment by SpriterGors** [comment]
I mentioned this issue here https://github.com/bluesky-social/social-app/issues/5764, and also is mentioned in https://github.com/bluesky-social/social-app/issues/6920 and https://github.com/bluesky-social/social-app/issues/5885

If the automated detector doesn't kick in reliably, at least we should have a more robust means to select the language before we hit "post"

**Comment by KernelLeak** [comment]
Most of the time this happens to me, I'm actually replying to a post in German while language is set to English, or vice versa (and the detector not kicking in) - a very easy solution to fix most of these cases would be to default to the language of the post that's being replied to when replying.

**Comment by surfdude29** [comment]
> […] a very easy solution to fix most of these cases would be to default to the language of the post that's being replied to when replying.

I like this idea, although I have a couple of thoughts:
- A toast or some kind of pop-up should let the user know that the post language has been changed
- I think it should only be changed for that reply, and should then return to the previous setting

> Oh, and actually being able to see the language a post is tagged as anywhere would help noticing it after posting...

This would definitely be useful, perhaps the Translate button could be changed to show the language, e.g. instead of just **Translate** it could say **Translate from [language name]**.

Also, in case you're curious about a particular post, you can look at the record on PDSls to check what language it's tagged as:

https://pdsls.dev/

**Comment by KernelLeak** [comment]
> * A toast or some kind of pop-up should let the user know that the post language has been changed

Agreed.

> * I think it should only be changed for that reply, and should then return to the previous setting

That was actually my intention - do this only for replies; keep the last used language for new standalone posts.

> This would definitely be useful, perhaps the Translate button could be changed to show the language, e.g. instead of just **Translate** it could say **Translate from [language name]**.

This still hides the info in a menu; maybe it could just have a small badge with the 2-letter ISO code?

> Also, in case you're curious about a particular post, you can look at the record on PDSls to check what language it's tagged as:

Since I use BlueSky strictly in a web browser I'm just digging through the JSON response in the network tab of DevTools; it's probably quicker that way.

**Comment by surfdude29** [comment]
> Most of the time this happens to me, I'm actually replying to a post in German while language is set to English, or vice versa (and the detector not kicking in) - a very easy solution to fix most of these cases would be to default to the language of the post that's being replied to when replying.

There's an update on this language-related issue discussed earlier (although the core issue of language detection being unreliable would not be impacted).

When merged, this PR will mean that a user who is replying to a post that has a language code different to the one they currently have selected will immediately get a prompt suggesting to change the post language setting to match:

- #8832

**Comment by ssg** [comment]
The text below isn't detected as English:

> visual studio 2026 has been released as just "visual studio". that's a significant change as it's always been branded with a year for every release since 2002. the year is still there on most UI elements, but it feels like microsoft is phasing it out.

**Comment by ssg** [comment]
The text below isn't detected as English:

> as i understand, linux lacks a “ready to setup” state (like iOS and windows first time welcome and configuration aka OOBE screens). 

> for instance, i can’t donate an installed linux machine ready to configure with a few clicks. people must install it from scratch.

**Comment by ssg** [comment]
this wasn't detected as Turkish:

> tamam bunda da hayvan gibi radyatör var. mantık şu benim anladığım: fanın sessiz olması için yavaş devirle iş görebilmesi lazım. onun için de sıcaklığın fana kadar hızlı taşınabilmesi ve daha dağıtılması lazım ki fana daha az iş düşsün.

**Comment by SpriterGors** [comment]
I don't know how the language detection algorithm works, but in those cases, shouldn't it activate when certain letters and clusters are present in the language? For example in the Turkish example above, the frequent presence of the dotless i, ç and ş should have at least activated the prompt somehow.

The same can be said of English and words like "the", "and", "its", "you" etc.

[edit] And make it so that the algorithm only considers the languages you speak, so that it doesn't need to compare the written text with all existing languages.

**Comment by ssg** [comment]
@SpriterGors I agree that it should approach to language detection logically like that, but the existing tech is just a neural network trained on a corpus. That means, we don't even know what it takes into account. It might be biased towards certain languages based on the distribution of the corpus too.

Besides, it's common to write Turkish without any Turkish letters at all, especially for Turkish people living abroad. So, letters aren't enough to distinguish the language. Sometimes, I write a Turkish sentence with lots of English words especially on English topics, but the grammar is still Turkish. All those things need to be taken into account.
